cmd/internal/cli/pull.go

```diff
@@ -174,6 +174,12 @@ func pullRun(cmd *cobra.Command, args []string) {
 	case HTTPProtocol, HTTPSProtocol:
 		libexec.PullNetImage(name, args[i], force)
 	default:
+		if !force {
```

Comment:
So, I think as a fix for this issue this PR is fine. But at the same time, the way we're handling `--force` looks like complete and utter spaghetti code across the board. We're replicating this code block 2 times, but passing `force` to `PullShubImage` and `PullNetImage`. Not to mention that `libexec` as a package is meaningless and should *really* just be removed altogether. Those functions (and the rest of the code in this switch statement) should seriously only live in `internal/app/singularity`

Reply:
Agreed, should be something to be refactored in the future